### ISSUE TITLE
transform_graph tools fails to produce output for a particular graph definition/also strange behavior of the tool

### ISSUE THREAD

**Issue by drsealks** [tensorflow/tensorflow#26867]
<em>Please make sure that this is a bug. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):
No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):
Linux Ubuntu 16.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
n/a
- TensorFlow installed from (source or binary):
source
- TensorFlow version (use command below):
v1.12.0-10232-g9a43dfe 1.13.1
- Python version:
3.7.1
- Bazel version (if compiling from source):
Build label: 0.19.2
Build target: bazel-out/k8-opt/bin/src/main/java/com/google/devtools/build/lib/bazel/BazelServer_deploy.jar
Build time: Mon Nov 19 16:25:09 2018 (1542644709)
Build timestamp: 1542644709
Build timestamp as int: 1542644709
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version:
n/a
- GPU model and memory:
n/a

You can collect some of this information using our environment capture [script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with
python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"

**Describe the current behavior**
I was trying to use `transform_graph` tool to produce quantized version of a very simple graph. I noticed that in certain configurations of the graph(they are simple structurally) the tools complains about missing ops and so on. An example message:
```
2019-03-19 07:55:57.750401: W tensorflow/tools/graph_transforms/transform_utils.cc:441] Expected MobilenetV1/conv1/batch_normalization_v1/beta to be preserved.
2019-03-19 07:55:57.750414: W tensorflow/tools/graph_transforms/transform_utils.cc:441] Expected MobilenetV1/conv1/batch_normalization_v1/gamma to be preserved.
2019-03-19 07:55:57.750423: W tensorflow/tools/graph_transforms/transform_utils.cc:448] Generator function didn't preserve needed nodes, copying old replacements back in instead.
```
I tried to create a minimalistic reproducible example, where I started to notice more strange behavior. I will attach the python code for graph_def building below. For now this is the pb file I'm trying to work with:
[saved_model.pb.tar.gz](https://github.com/tensorflow/tensorflow/files/2982144/saved_model.pb.tar.gz)

In my setup the tools fails with the following error:
`2019-03-19 07:59:51.381714: E tensorflow/tools/graph_transforms/transform_utils.cc:577] Invalid input batch_normalization/gamma for node batch_normalization/FusedBatchNorm - name: "batch_normalization/FusedBatchNorm"`

I call the the tool with the following configuration:
`$tbin --in_graph=$in_graph --out_graph=$out_graph_clean_quant --inputs=$inputs --outputs=$outputs --transforms='add_default_attributes strip_unused_nodes() remove_nodes(op=Identity, op=CheckNumerics)  quantize_weights quantize_nodes fold_batch_norms fold_old_batch_norms strip_unused_nodes sort_by_execution_order'`

The tool will not fail if we change the depthwise conv to a normal conv2d. FusedBatchNorm does not lose any inputs, it just works. 

Another observation: if we remove the following transformations `strip_unused_nodes() remove_nodes(op=Identity, op=CheckNumerics)` the tool will not fail.

Another observation. If we remove two aforementioned transformations, this is a piece of graph we get:
![image](https://user-images.githubusercontent.com/6204851/54589454-940e8f80-4a1d-11e9-946f-a24d220908e3.png)

For some reason two FusedBatchNorm nodes receive Beta\Gamma from the same nodes. I'm guessing this should not be the case. Furthermore, inside each of this nodes gamma\variance and beta\mean pairs receive values from the same reading ops. Note that there are only two additional inputs to FusedBatchNorm.
![image](https://user-images.githubusercontent.com/6204851/54589580-ecde2800-4a1d-11e9-9679-b58f3b963735.png)

Hopefully it all makes sense.

**Describe the expected behavior**
Tool should not fail when FusedBatchNorm is followed by DepthwiseConv2dNative.

**Code to reproduce the issue**
Here is some python self-contained code to reproduce the problem:

```
from tensorflow.contrib import keras
import tensorflow as tf
import os


def freeze_graph(model_dir, checkpoint_name, output_node_names):
    """Extract the sub graph defined by the output nodes and convert
    all its variables into constant

    Args:
        model_dir: the root folder containing the checkpoint state file
        output_node_names: a string, containing all the output node's names,
                            comma separated
    """
    if not tf.gfile.Exists(model_dir):
        raise AssertionError(
            "Export directory doesn't exists. Please specify an export "
            "directory: %s" % model_dir)

    if not output_node_names:
        print("You need to supply the name of a node to --output_node_names.")
        return -1

    absolute_model_dir = os.path.split(checkpoint_name)[0]
    output_graph = os.path.join(absolute_model_dir, "saved_model.pb")

    clear_devices = True

    # We start a session using a temporary fresh Graph
    with tf.Session() as sess:
        sess.run(tf.global_variables_initializer())
        saver = tf.train.Saver()
        saver.restore(sess, checkpoint_name)
        output_graph_def = tf.graph_util.convert_variables_to_constants(
            sess,  # The session is used to retrieve the weights
            tf.get_default_graph().as_graph_def(),  # The graph_def is used to retrieve the nodes
            output_node_names  # The output node names are used to select the usefull nodes
        )
        with tf.gfile.GFile(output_graph, "wb") as f:
            f.write(output_graph_def.SerializeToString())

        with tf.gfile.GFile(output_graph, "rb") as f:
            output_graph_def.ParseFromString(f.read())
        print("%d ops in the final graph." % len(output_graph_def.node))


save_folder = os.path.join(os.path.expanduser('~'), 'tmp')
os.makedirs(save_folder, exist_ok=True)

initializer = tf.contrib.layers.xavier_initializer_conv2d()
wdw = tf.get_variable(shape=(3, 3, 32, 1), initializer=initializer, name="weights")
wc = tf.get_variable(shape=(1, 1, 32, 1), initializer=initializer, name="weights1")

conv_out = tf.placeholder(shape=(1, 300, 300, 3), dtype=tf.float32)
conv_out = tf.nn.conv2d(conv_out, tf.get_variable(shape=(1, 1, 3, 32), initializer=initializer,name='w1'),
                        strides=[1, 1, 1, 1],
                        padding='SAME')
conv_out = keras.layers.BatchNormalization()(conv_out, training=False)

conv_out = tf.nn.depthwise_conv2d(conv_out, wdw, strides=[1, 1, 1, 1], padding='SAME')
# conv_out = tf.nn.conv2d(conv_out, tf.get_variable(shape=(1, 1, 32, 32), initializer=initializer, name="weights32"), strides=[1, 1, 1, 1], padding='SAME')
conv_out = keras.layers.BatchNormalization()(conv_out, training=False)
conv_out = tf.nn.relu(conv_out)
#
conv_out = tf.nn.conv2d(conv_out, wc, strides=[1, 1, 1, 1], padding='SAME')
conv_out = keras.layers.BatchNormalization()(conv_out, training=False)
conv_out = tf.nn.relu(conv_out)

saver = tf.train.Saver()
checkpoint_name = os.path.join(save_folder, 'model.chkpt')
with tf.Session() as s:
    s.run(tf.global_variables_initializer())

    save_path = saver.save(s, checkpoint_name)

freeze_graph(save_folder, checkpoint_name, [conv_out.op.name])
print(conv_out.op.name)
```

**Other info / logs**
Please, do let me know if I missed something. 
Also, I might be misunderstanding something - please let me know if this is the case

**Comment by tensorflowbutler** [comment]
Hi There,

 We are checking to see if you still need help on this issue, as you are using an older version of tensorflow(1.x) which is officially considered as end of life. We recommend that you upgrade to 2.4 or later version and let us know if the issue still persists in newer versions. 

 This issue will be closed automatically 7 days from now. If you still need help with this issue, Please open a new issue for any help you need against 2.x, and we will get you the right help.
